### PR TITLE
Elemental Analysis - Plotting Interface Tests

### LINKED ISSUE #23315: Elemental Analysis - Plotting Interface Testing
Follows on from #23313.

Adds unit testing for the plotting interface.

### PR DESCRIPTION
**Pull request by ewancook**
*Merge after #23313*

**Description of work.**
Adds tests for #23313

**Report to:** Aidy

**To test:**
Automatic as part of CI (unit testing)

Closes  #23315

*This does not require release notes* because **they were added in a previous PR**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/PlottingView_test.py [hunk 1]
```diff
@@ -1,8 +1,445 @@
...
+                          self.plots_return_value, self.mock_workspace])
+
+    def test_add_workspace_name_if_in_workspaces(self):
+        self.view.workspaces = {}
+        self.view._add_workspace_name(self.plot_name, self.mock_workspace)
+        self.assertEquals(self.view.workspaces[self.plot_name], [
+                          self.mock_workspace])
+
+
+class PlottingViewPlotFunctionsTests(unittest.TestCase):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/PlottingView_test.py]: whats the difference between this and the class above?

Comment (ewancook) [reply scripts/test/Muon/PlottingView_test.py]: There are two reasons I remember: 
- To split them up by functionality
- The second class has a different set of methods and variables that need to be mocked that don't necessarily overlap with the first (and would mean having to unmock functions when called)

Comment (AnthonyLim23) [reply scripts/test/Muon/PlottingView_test.py]: could we have it in its own file? It is rather long with the two of them

Comment (ewancook) [reply scripts/test/Muon/PlottingView_test.py]: Can do

**Comment by ewancook** [comment]
Tests will be rewritten after plotting changes from #23426 are finished

File: scripts/test/Muon/PlottingView_test.py [hunk 2]
```diff
@@ -69,42 +72,15 @@ def setUp(self):
 
     def test_redo_layout_with_plots_not_equal_to_zero(self):
         self.view.plots = [mock.Mock() for i in range(3)]
-        self.view.call_plot_method(
-            self.mock_name,
-            self.mock_func,
-            *self.mock_args)
-        self.mock_func.assert_called_once_with(*self.mock_args)
```
Comment (AnthonyLim23) [inline scripts/test/Muon/PlottingView_test.py]: can we have a comment on why movable v lines

File: scripts/test/Muon/PlottingView_test.py (hunk 1 above)
Comment (AnthonyLim23) [reply scripts/test/Muon/PlottingView_test.py]: see issue #23498

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please?

**Pull request merged**